regression-tests/pure2-bugfix-for-deducible-parameters.cpp2

```diff
@@ -0,0 +1,34 @@
+// Dependent, non-deducible parameters
+// are wrapped like non-dependent parameters.
+init: <T> (out x: std::integral_constant<i32, T::value>) = { x = (); }
+init: <T> (out x: std::integral_constant<i32, T::value>, _: T) = { x = (); }
+id: <T> (x: std::integral_constant<i32, T::value>) -> forward _ = x;
+id: <T> (x: std::integral_constant<i32, T::value>, y: T) = { [[assert: x& == y&]] }
+
+main: () = {
+  zero: type == std::integral_constant<i32, 0>;
+
+  z: zero;
+  init<zero>(out z);
+  [[assert: id<zero>(z)& == z&]]
+
+  // Deducible parameters.
+  _ = :v = 0;
+  _ = :<T> (x: std::vector<T>) = {}(:std::vector<i32> = ());
+  _ = :<T> (x: std::vector<std::vector<T>>) = {}(:std::vector<std::vector<i32>> = ());
+  // Uncomment once `typename` is supported for template arguments.
+  // _ = :<T, U> (x: std::pair<T, typename U::value_type>, y: U) = {}(:std::pair = (0, 0), z);
+  _ = :<T, U> (x: std::array<T, U::value>, y: U) = {}(:std::array<i32, 0> = (), z);
+  init(out z, z);
```

Comment:
These `_ =` are not needed, right?

Reply:
And similarly throughout the other new test file below?

Reply:
That's right.
It's just to prevent unused-things warnings.